### PR TITLE
Sync -release-blocking dashboards with kubernetes/release/lib/releaselib.sh

### PR DESCRIPTION
**Pull request by david-mcmahon**
This is the first step in [centralizing test suite](https://github.com/kubernetes/release/issues/340) definitions.  

~~This splits the current release blocking lists into hard (binary signal) and soft (human ack required) and provides a place for [releaselib.sh](https://github.com/kubernetes/release/blob/master/lib/releaselib.sh#L104) to consume the list of suites rather than hardcoding them leading to them inevitably drifting out of sync.~~

~~The "-hard" lists in this PR are a best guess based on a union of the "-soft" lists and what's currently in [releaselib.sh](https://github.com/kubernetes/release/blob/master/lib/releaselib.sh#L104).~~

~~I do not expect this is the final list.  Release managers, please chime in here.  This is to help you with a simple 'gate' to provide release signal.  Ideally we'd have everything in one list vs hard/soft.  We'll get there, but in the meantime you'll use both lists to get a reasonable 'hard' signal and a human ack by checking the soft list in conjunction.~~

This simply syncs up the "blocking" suites with what is currently in [releaselib.sh](https://github.com/kubernetes/release/blob/master/lib/releaselib.sh#L104).

Fixes https://github.com/kubernetes/release/pull/340

cc @kubernetes/kubernetes-release-managers @dclaar @javier-b-perez

### CODE REVIEW

File: testgrid/config/config.yaml [hunk 1]
```diff
@@ -1990,8 +1990,40 @@ dashboards:
   - name: periodic-kubeadm-gce-1.6
     test_group_name: periodic-kubernetes-e2e-kubeadm-gce-1-6
 
+# These are the release *blocking* tests.  These provide a valid binary signal
+# to gate releases (alpha, beta, official).
+# This list is used by kubernetes/release/lib/releaselib.sh to provide no/nogo
```
Comment (enisoc) [inline testgrid/config/config.yaml]: This seems a bit pessimistic. We do sometimes get a `go` signal.

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -1990,8 +1990,40 @@ dashboards:
...
+  - name: gke-release-1-6
+    test_group_name: ci-kubernetes-e2e-gke-release-1-6
+  - name: gke-slow-release-1-6
+    test_group_name: ci-kubernetes-e2e-gke-slow-release-1-6
+
+# These are the release "blocking" tests.
+# These need to be interpreted by a human
 # Matches the list on https://github.com/kubernetes/test-infra/issues/2029
-- name: release-1.6-blocking
+- name: release-1.6-blocking-hard-soft
```
Comment (enisoc) [inline testgrid/config/config.yaml]: Should be just `-soft`?

File: testgrid/config/config.yaml (hunk 1 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: How do you mean pessimistic?  Sometimes we get a 'go' signal from what?  The current go/no go is using the list in releaselib.sh.

**Comment by enisoc** [comment]
This sounds like a step in the right direction to me. One concern I have is that after `anago` fetches the list from somewhere centralized, we probably need a flag to bypass certain broken hard-blocker tests on a case-by-case basis. We do this often for alphas, less often for betas, and... as for official releases I will plead the fifth.

In the past, I've accomplished that by temporarily commenting out a line in `releaselib.sh`, but that won't work anymore if it downloads the list.

File: testgrid/config/config.yaml (hunk 1 above)
Comment (enisoc) [inline testgrid/config/config.yaml]: Sorry I was being too cute. There is a typo that says the two signals are `no` or `nogo`.

**Comment by david-mcmahon** [comment]
@enisoc That is addressed in https://github.com/kubernetes/release/pull/341.

File: testgrid/config/config.yaml [hunk 3]
```diff
@@ -1990,8 +1990,40 @@ dashboards:
   - name: periodic-kubeadm-gce-1.6
     test_group_name: periodic-kubernetes-e2e-kubeadm-gce-1-6
 
+# These are the release *blocking* tests.  These provide a valid binary signal
+# to gate releases (alpha, beta, official).
+# This list is used by kubernetes/release/lib/releaselib.sh to provide no/nogo
+# signal for releases.
+# The first test/suite in the list is the primary suite for release purposes
+- name: release-1.6-blocking-hard
```
Comment (ixdy) [inline testgrid/config/config.yaml]: maybe leave the name of the "blocking-hard" groups as-is, so we don't break bookmarks?

File: testgrid/config/config.yaml [hunk 4]
```diff
@@ -1990,8 +1990,40 @@ dashboards:
...
+  - name: gke-1-6
+    test_group_name: ci-kubernetes-e2e-gke-release-1-6
+  - name: gke-slow-1-6
+    test_group_name: ci-kubernetes-e2e-gke-slow-release-1-6
+
+# These are the release "blocking" tests.
+# These need to be interpreted by a human
 # Matches the list on https://github.com/kubernetes/test-infra/issues/2029
-- name: release-1.6-blocking
+- name: release-1.6-blocking-soft
```
Comment (ixdy) [inline testgrid/config/config.yaml]: what's the difference between the "soft" blocking tests and the rest of the tests that are not blocking in any way? why do we need to differentiate them?

Comment (david-mcmahon) [inline testgrid/config/config.yaml]: I suppose we don't?  The "release blocking" set came about as a way for @kubernetes/kubernetes-release-managers to keep an eye on "release blocking" tests, however, that list did not take into account the existing *actual* set of "release blocking" tests defined in [releaselib.sh](https://github.com/kubernetes/release/blob/master/lib/releaselib.sh#L104).

Ideally we'd have ONE set of release-blocking tests (per release) that were 1) not flaky 2) can produce a true go/nogo gate for a release.

Comment (david-mcmahon) [inline testgrid/config/config.yaml]: Maybe I should just run with that and redefine blocking to be exactly what it should be.  The other tests that folks want to eyeball for whatever reason can be eyeballed elsewhere.  @kubernetes/kubernetes-release-managers can I get an ack on that so that this doesn't get reverted as soon as it goes through?  Let's decide now.

**Comment by david-mcmahon** [comment]
The split idea doesn't seem to have much traction.  Let's see how everyone likes just getting the -release-blocking dashboards in sync with what have actually been the release blocking tests for a year and a half now.

**Comment by david-mcmahon** [comment]
@kubernetes/kubernetes-release-managers I know this is something you're currently working around.  It would be good to get this in sooner than later.  Please have a look and lgtm.

File: testgrid/config/config.yaml [hunk 5]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
     test_group_name: periodic-kubernetes-bazel-test-1-7
 
+# These are the release *blocking* tests.  These provide a valid binary signal
+# to gate releases (alpha, beta, official).
+# This list is used by kubernetes/release/lib/releaselib.sh to provide no/nogo
+# signal for releases.
+# The first test/suite in the list is the primary suite for release purposes
 - name: release-1.7-blocking
   dashboard_tab:
-  - name: build-1.7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: Let's keep the build test? Actually i think the build test should be the primary test. cc @krzyzacy

File: testgrid/config/config.yaml [hunk 6]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
     test_group_name: ci-kubernetes-test-go-release-1.7
   - name: verify-1.7
     test_group_name: ci-kubernetes-verify-release-1.7
   - name: kubelet-1.7
     test_group_name: ci-kubernetes-node-kubelet-1.7
-  - name: federation-build-1.7
-    test_group_name: ci-kubernetes-federation-build-1.7
-  - name: gce-release-1-7
-    test_group_name: ci-kubernetes-e2e-gce-release-1-7
-  - name: gci-gce-release-1-7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: @dchen1107 do we need all the gci-, gke-, gci-gke- version of all the tests?

File: testgrid/config/config.yaml [hunk 7]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
-    test_group_name: ci-kubernetes-e2e-gci-gce-release-1-7
+  - name: gke-serial-release-1-7
+    test_group_name: ci-kubernetes-e2e-gke-serial-release-1-7
   - name: gke-release-1-7
     test_group_name: ci-kubernetes-e2e-gke-release-1-7
-  - name: gci-gke-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gke-release-1-7
-  - name: aws-release-1-7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: Let's keep one aws suite.

File: testgrid/config/config.yaml [hunk 8]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
-    test_group_name: ci-kubernetes-e2e-gci-gke-slow-release-1-7
-  - name: gce-serial-release-1-7
-    test_group_name: ci-kubernetes-e2e-gce-serial-release-1-7
-  - name: gci-gce-serial-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gce-serial-release-1-7
-  - name: gke-serial-release-1-7
-    test_group_name: ci-kubernetes-e2e-gke-serial-release-1-7
-  - name: gci-gke-serial-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gke-serial-release-1-7
-  - name: gce-federation-release-1-7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: let's keep a federation test.

File: testgrid/config/config.yaml [hunk 9]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
-    test_group_name: ci-kubernetes-e2e-gce-federation-release-1-7
-  - name: gce-alpha-features-release-1-7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: let's keep the alpha features, thought it's failing

File: testgrid/config/config.yaml [hunk 10]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
-    test_group_name: ci-kubernetes-e2e-gce-alpha-features-release-1-7
-  - name: gci-gce-alpha-features-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gce-alpha-features-release-1-7
-  - name: gce-scalability-release-1-7
-    test_group_name: ci-kubernetes-e2e-gce-scalability-release-1-7
-  - name: gci-gce-scalability-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gce-scalability-release-1-7
-  - name: gci-gce-ingress-release-1-7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: @dchen1107 do we need the ingress test?

File: testgrid/config/config.yaml [hunk 11]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
-    test_group_name: ci-kubernetes-e2e-gci-gke-ingress-release-1-7
-  - name: gce-reboot-release-1-7
-    test_group_name: ci-kubernetes-e2e-gce-reboot-release-1-7
-  - name: gci-gce-reboot-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gce-reboot-release-1-7
-  - name: gke-reboot-release-1-7
-    test_group_name: ci-kubernetes-e2e-gke-reboot-release-1-7
-  - name: gci-gke-reboot-release-1-7
-    test_group_name: ci-kubernetes-e2e-gci-gke-reboot-release-1-7
-  - name: soak-gce-1-7-deploy
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: Do we need the soak test? They are all failing.

File: testgrid/config/config.yaml [hunk 12]
```diff
@@ -2674,78 +2581,35 @@ dashboards:
...
-    test_group_name: ci-kubernetes-soak-gce-1-7-deploy
-  - name: soak-gce-1-7-test
-    test_group_name: ci-kubernetes-soak-gce-1-7-test
-  - name: soak-gci-gce-1-7-deploy
-    test_group_name: ci-kubernetes-soak-gci-gce-1-7-deploy
-  - name: soak-gci-gce-1-7-test
-    test_group_name: ci-kubernetes-soak-gci-gce-1-7-test
-  - name: gce-gpu-1-7
-    test_group_name: ci-kubernetes-e2e-gce-gpu-1-7
-  - name: gce-kubeadm-1.7
```
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: let's keep this.

File: testgrid/config/config.yaml (hunk 5 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: Thanks for the replies, @caesarxuchao!  That does seem a reasonable primary test suite.  The current list of suites hasn't been updated in some months and many new suites have been created, so a full audit is due.  If noone objects, we'll make the build job primary.

File: testgrid/config/config.yaml (hunk 6 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: Sure?  This is up to all of you @kubernetes/kubernetes-release-managers.  Remember the goal here when coming up with a set of HARD gating tests, is that you can get a passing set with some regularity as the point is to block the release.  Don't add anything here that's just "been failing forever".  And the suites may differ between branches (master vs. release-*).

File: testgrid/config/config.yaml (hunk 7 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: OK, for master and *all* "blocking" sets?

File: testgrid/config/config.yaml (hunk 8 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: OK, for master and all "blocking" sets?

File: testgrid/config/config.yaml (hunk 9 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: That's what you probably don't want then in a blocking set?  You can always check other tabs/dashboards for other "suites of interest", but something that's just always red doesn't belong here.

File: testgrid/config/config.yaml (hunk 8 above)
Comment (enisoc) [inline testgrid/config/config.yaml]: Only keep federation for master and 1.7+. For 1.6 and lower, federation test is permanently broken and WNF.

File: testgrid/config/config.yaml [hunk 13]
```diff
@@ -1990,159 +1990,66 @@ dashboards:
   - name: periodic-kubeadm-gce-1.6
     test_group_name: periodic-kubernetes-e2e-kubeadm-gce-1-6
 
-# Matches the list on https://github.com/kubernetes/test-infra/issues/2029
+# These are the release *blocking* tests.  These provide a valid binary signal
+# to gate releases (alpha, beta, official).
+# This list is used by kubernetes/release/lib/releaselib.sh to provide no/nogo
```
Comment (enisoc) [inline testgrid/config/config.yaml]: Should be `go/nogo`.

File: testgrid/config/config.yaml [hunk 14]
```diff
@@ -1990,159 +1990,66 @@ dashboards:
...
 - name: release-1.6-blocking
   dashboard_tab:
-  # Build/Verify
-  - name: build-1.6
-    test_group_name: ci-kubernetes-build-1.6
-  - name: verify-1.6
-    test_group_name: ci-kubernetes-verify-release-1.6
-  - name: test-go-1.6
-    test_group_name: ci-kubernetes-test-go-release-1.6
-  - name: federation-build-1.6
```
Comment (enisoc) [inline testgrid/config/config.yaml]: Could you move this to release-1.6-all? It's not there yet and I don't want it to fall out entirely.

File: testgrid/config/config.yaml [hunk 15]
```diff
@@ -1990,159 +1990,66 @@ dashboards:
...
-  # E2E
   - name: gce-1.6
     test_group_name: ci-kubernetes-e2e-gce-release-1-6
-  - name: gci-gce-1.6
-    test_group_name: ci-kubernetes-e2e-gci-gce-release-1-6
-  - name: gke-1.6
-    test_group_name: ci-kubernetes-e2e-gke-release-1-6
-  - name: gci-gke-1.6
-    test_group_name: ci-kubernetes-e2e-gci-gke-release-1-6
-  - name: kubeadm-gce-1.6
```
Comment (enisoc) [inline testgrid/config/config.yaml]: Could you move this to release-1.6-all? It's not there yet and I don't want it to fall out entirely.

**Comment by david-mcmahon** [comment]
Thanks @caesarxuchao and @enisoc. All comments addressed. PTAL.

File: testgrid/config/config.yaml (hunk 10 above)
Comment (dchen1107) [inline testgrid/config/config.yaml]: Yes

File: testgrid/config/config.yaml (hunk 6 above)
Comment (dchen1107) [inline testgrid/config/config.yaml]: Currently we need all of them. But in long term, I wish we can remove gke related or treat them differently in gke release cycle.

File: testgrid/config/config.yaml (hunk 12 above)
Comment (dchen1107) [inline testgrid/config/config.yaml]: +1

File: testgrid/config/config.yaml [hunk 16]
```diff
@@ -2676,78 +2591,45 @@ dashboards:
   - name: periodic-bazel-test-1.7
     test_group_name: periodic-kubernetes-bazel-test-1-7
 
+# These are the release *blocking* tests.  These provide a valid binary signal
+# to gate releases (alpha, beta, official).
+# This list is used by kubernetes/release/lib/releaselib.sh to provide go/nogo
+# signal for releases.
+# The first test/suite in the list is the primary suite for release purposes
 - name: release-1.7-blocking
```
Comment (dchen1107) [inline testgrid/config/config.yaml]: Can we add three more:

1) kubelet-serial-gce-e2e-1-7
2) gke-gci-gpu-1-7
3) gce-gci-auth-1-7

Comment (david-mcmahon) [inline testgrid/config/config.yaml]: These looks like brand new variants, not just something to move into -blocking.  That seems to be outside the scope of this PR.  Post-this, anyone can add those new suites and put them into blocking.

File: testgrid/config/config.yaml (hunk 6 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: *All* of those?  Can you be more specific?  There are LOTS of suites that fit the *gci* variant.  Are all of those passing enough to provide a good signal at this point? Should whatever that list ends up being be the same for 1.7 and master (and 1.6 for that matter)?

**Comment by david-mcmahon** [comment]
@dchen1107 @caesarxuchao How about we get this in and iterate as needed?  I think this is close enough and it sounds like things are pretty fluid still anyway.

**Comment by spiffxp** [comment]
I am slightly weirded out that we're using a config meant for testgrid as the canonical source for release tools.  What am I missing?

File: testgrid/config/config.yaml (hunk 6 above)
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: I mean all gci-,gke-,gci-gke- tests that were in the 1.7-blocking should remain there.

File: testgrid/config/config.yaml (hunk 16 above)
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: If it's not difficult, let's just add them in this PR.

**Comment by caesarxuchao** [comment]
@david-mcmahon i'm using the 1.7-blocking panel to monitor the branch health every day so i hope we can get it right in one iteration. Please ping me if any comment is unclear.

File: testgrid/config/config.yaml (hunk 16 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: Added these.  As these are brand new test suites and not just moving things around, I have little confidence I did it correctly.  Please check carefully.

Comment (caesarxuchao) [inline testgrid/config/config.yaml]: Sorry, none of these tests exists:
https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/kubelet-serial-gce-e2e-1-7
https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-auth-1-7
https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-gpu-1-7

Let's leave them in the TODO. @dchen1107 who should create these tests?

**Comment by fejta** [comment]
@spiffxp it does seem a little odd but on the other hand a) it is what teams use to monitor and b) we cannot add/remove jobs without adding/removing them from testgrid. So it is probably the best place for this to live for now.

**Comment by david-mcmahon** [comment]
@spiffxp @fejta This PR answers an immediate need based on where the data is currently stored.  Could these lists be stored elsewhere? Sure.  If you have thoughts/suggestions on how you'd prefer to do this, open a new issue and let's move it.

For now this at least syncs up very important lists that provide a key measurement of release health. @kubernetes/kubernetes-release-managers if this looks good, please lgtm so we can move this forward.

**Comment by caesarxuchao** [comment]
@david-mcmahon it looks like the net change for 1.7-blocking is removed the gpu test? Can we add it back?

File: testgrid/config/config.yaml (hunk 16 above)
Comment (caesarxuchao) [inline testgrid/config/config.yaml]: @david-mcmahon let's add these three into the panel. @dchen1107 says there are PRs creating these tests.

**Comment by fejta** [comment]
I think it is a good start how it is written now. We have all the unit tests in place -- we can't stop running jobs without also deleting them from testgrid. We can't add new jobs without adding them to testgrid somewhere. 

If we do add a list somewhere we need to make a unit test to ensure it matches the jobs in testgrid since these are what people monitor on a daily basis.

File: testgrid/config/config.yaml (hunk 16 above)
Comment (david-mcmahon) [inline testgrid/config/config.yaml]: I can't add them until the PR goes in that includes them.  Let's just get this in now.  We can iterate.

**Comment by caesarxuchao** [comment]
@david-mcmahon could you add back the gpu test to 1.7 blocking? Then lgtm. Thanks for bearing with me.

**Comment by david-mcmahon** [comment]
@krzyzacy We're probably colliding here.  Please check these changes.

File: testgrid/config/config.yaml [hunk 17]
```diff
@@ -2053,168 +2055,73 @@ dashboards:
     test_group_name: periodic-kubernetes-bazel-build-1-6
   - name: periodic-kubeadm-gce-1.6
     test_group_name: periodic-kubernetes-e2e-kubeadm-gce-1-6
+  - name: -kubeadm-gce-1.6
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: nit: maybe `ci-kubeadm-gce-1.6`, or it's actually `postsubmit-kubeadm-gce-1.6`

**Comment by krzyzacy** [comment]
@david-mcmahon agree we can iterate, we can merge this first and I'll make follow-up PRs for other 1.7 blocking jobs.
